sklearn/datasets/samples_generator.py

```diff
@@ -379,14 +379,16 @@ def sample_example():
     if not sparse:
         X = X.toarray()
 
-    if return_indicator:
-        lb = MultiLabelBinarizer()
+    # return_indicator can be True due to backward compatibility
+    if (return_indicator == True or return_indicator == 'dense'
+                                 or return_indicator == 'sparse'):
```

Comment:
Non-standard indentation. For one line, could do `if return_indicator in (True, 'spare', 'dense'):` or just `if return_indicator:`